### PR TITLE
Decouple database dependencies from the executor

### PR DESCRIPTION
**Pull request by christyjacob4**
## What does this PR do?

Decouple the executor from Appwrite. 

## Test Plan
Use existing tests

## Related PRs and Issues
NA

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

YES

### CODE REVIEW

**Comment by christyjacob4** [comment]
@eldadfux  I've removed all the database occurrences from the executor except the one in the shut down method
https://github.com/appwrite/appwrite/pull/2739/files#diff-bc51487c1b7afebab7023312e35731a490f9ea0dc03619acd859f384d819a0f0R790

This is responsible for deleting ongoing executions and updating their status as failed in the database. 
Let me know if you have any insights.

File: app/controllers/api/functions.php [hunk 1]
```diff
@@ -769,6 +769,16 @@
             throw new Exception('Deployment not found. Deploy deployment before trying to execute a function', 404);
```
Comment (eldadfux) [inline app/controllers/api/functions.php:769]: Deploy a deployment sound weird, can we switch to create a deployment?

File: app/controllers/api/functions.php [hunk 2]
```diff
@@ -817,32 +827,59 @@
...
+            'deploymentId' => $deployment->getId(),
+            'buildId' => $deployment->getAttribute('buildId', ''),
+            'path' => $build->getAttribute('outputPath', ''),
+            'vars' => $function->getAttribute('vars', []), 
             'data' => $data,
+            'runtime' => $function->getAttribute('runtime', ''),
+            'timeout' => $function->getAttribute('timeout', 0),
+            'baseImage' => '',
             'webhooks' => $project->getAttribute('webhooks', []),
             'userId' => $user->getId(),
```
Comment (eldadfux) [inline app/controllers/api/functions.php:882]: Seems like we're missing a few values. jwt, trigger, projectId, executionId

File: app/executor.php [hunk 3]
```diff
@@ -7,6 +7,7 @@
 use Appwrite\Utopia\Response;
 use Appwrite\Utopia\Response\Model\Execution;
 use Cron\CronExpression;
+use LanguageServerProtocol\Range;
```
Comment (eldadfux) [inline app/executor.php:10]: What is LanguageServerProtocol?

File: app/controllers/api/functions.php [hunk 4]
```diff
@@ -817,32 +827,59 @@
...
+        //     'APPWRITE_FUNCTION_EVENT' => $event,
+        //     'APPWRITE_FUNCTION_EVENT_DATA' => $eventData,
+        //     'APPWRITE_FUNCTION_DATA' => $data,
+        //     'APPWRITE_FUNCTION_USER_ID' => $userId,
+        //     'APPWRITE_FUNCTION_JWT' => $jwt,
+        //     'APPWRITE_FUNCTION_PROJECT_ID' => $projectId
+        // ]);
+
         // Directly execute function.
         $ch = \curl_init();
```
Comment (eldadfux) [inline app/controllers/api/functions.php:869]: We could potentially abstract all this HTTP calls to the executor under: `src/Excutor/Excutor.php` until we have an SDK for it.

**Review by eldadfux — COMMENTED** [review]
Added a few initial comments. Not sure what are your plans, but if we can make the deletes worker unaware of the executor it could be nice.

File: app/controllers/api/functions.php (hunk 1 above)
Comment (christyjacob4) [reply app/controllers/api/functions.php:769]: I will update it 👍

File: app/controllers/api/functions.php (hunk 4 above)
Comment (christyjacob4) [reply app/controllers/api/functions.php]: done 👍

File: app/executor.php (hunk 3 above)
Comment (christyjacob4) [reply app/executor.php]: Hehhe imported it by mistake. Removed 👍

File: app/controllers/api/functions.php (hunk 2 above)
Comment (christyjacob4) [reply app/controllers/api/functions.php]: added them 👍

File: app/workers/functions.php [hunk 5]
```diff
@@ -168,92 +170,197 @@ function: $function,
...
+        /** Update execution status */
+        $execution->setAttribute('status', $executionResponse['status']);
+        $execution->setAttribute('statusCode', $executionResponse['statusCode']);
+        $execution->setAttribute('stdout', $executionResponse['stdout']);
+        $execution->setAttribute('stderr', $executionResponse['stderr']);
+        $execution->setAttribute('time', $executionResponse['time']);
+        $execution = Authorization::skip(fn() => $dbForProject->updateDocument('executions', $executionId, $execution));
+
+        /** Trigger Webhook */
+        $executionModel = new Execution();
```
Comment (lohanidamodar) [inline app/workers/functions.php:328]: let's follow similar pattern for naming. Probably the before use of `responseExecute` should be `executionResponse` ?

**Review by lohanidamodar — COMMENTED** [review]
Few queries and suggestions

File: app/workers/functions.php [hunk 6]
```diff
@@ -168,92 +170,197 @@ function: $function,
...
+            'APPWRITE_FUNCTION_EVENT_DATA' => $eventData,
+            'APPWRITE_FUNCTION_DATA' => $data,
+            'APPWRITE_FUNCTION_PROJECT_ID' => $projectId,
+            'APPWRITE_FUNCTION_USER_ID' => $userId,
+            'APPWRITE_FUNCTION_JWT' => $jwt,
+        ];
+        $vars = \array_merge($function->getAttribute('vars', []), $vars);
+
+        /** Execute function */
+        $executionResponse = $this->executor->createExecution(
```
[the remark this answers is not in the document]
Comment (christyjacob4) [reply app/workers/functions.php]: updated it 👍

File: app/workers/functions.php (hunk 5 above)
Comment (christyjacob4) [reply app/workers/functions.php:320]: changed both occurrences to `executionResponse ` 👍

**Pull request merged**